spec/02-integration/04-admin_api/22-debug_spec.lua

```diff
@@ -0,0 +1,460 @@
+        admin_listen = "127.0.0.1:9114",
+        cluster_listen = "127.0.0.1:9006",
+        admin_gui_listen = "off",
+        prefix = "cp2",
+        cluster_telemetry_listen = "localhost:9008",
+        log_level = "debug",
+      }))
+    end
+  end)
+
+  lazy_teardown(function()
+    helpers.stop_kong()
+    helpers.stop_kong("node2")
+
+    if strategy ~= "off" then
+      helpers.stop_kong("cp")
+      helpers.stop_kong("cp2")
+    end
+  end)
+
+  describe("/debug/{node, cluster}/log-level", function()
+    it("gets current log level for traditional and dbless", function()
+      local res = assert(helpers.admin_client():send {
+        method = "GET",
+        path = "/debug/node/log-level",
+      })
+
+      local body = assert.res_status(200, res)
+      local json = cjson.decode(body)
+      local message = "log level: debug"
+      assert(json.message == message)
+    end)
+
+    if strategy == "off" then
+      it("cannot change the log level for dbless", function()
+        local res = assert(helpers.admin_client():send {
+          method = "PUT",
+          path = "/debug/node/log-level/notice",
+        })
+        local body = assert.res_status(405, res)
+        local json = cjson.decode(body)
+        local message = "cannot change log level when not using a database"
+        assert(json.message == message)
+      end)
+      return
+    end
+
+    it("e2e test - check if dynamic set log level works", function()
+      local res = assert(helpers.admin_client():send {
+        method = "PUT",
+        path = "/debug/node/log-level/notice",
+      })
+
+      local body = assert.res_status(200, res)
+      local json = cjson.decode(body)
+      local message = "log level changed"
+      assert(json.message == message)
+
+      -- make sure we changed to notice
+      helpers.wait_until(function()
+        res = assert(helpers.admin_client():send {
+          method = "GET",
+          path = "/debug/node/log-level",
+        })
+        body = assert.res_status(200, res)
+        json = cjson.decode(body)
+        message = "log level: notice"
+        return json.message == message
+      end, 30)
+
+      -- go back to default (debug)
+      res = assert(helpers.admin_client():send {
+        method = "PUT",
+        path = "/debug/node/log-level/debug",
+      })
+      body = assert.res_status(200, res)
+      json = cjson.decode(body)
+      message = "log level changed"
+      assert(json.message == message)
+
+      -- make sure we changed to debug
+      helpers.wait_until(function()
+        res = assert(helpers.admin_client():send {
+          method = "GET",
+          path = "/debug/node/log-level",
+        })
+        body = assert.res_status(200, res)
+        json = cjson.decode(body)
+        message = "log level: debug"
+        return json.message == message
+      end, 30)
+
+      -- e2e test
+      helpers.clean_logfile()
+      assert.logfile().has.line("[debug]", true, 30)
+    end)
```

Comment:
Maybe this test can capture some error log, it will more complete

Reply:
Good idea, done